packages/react-component-library/src/components/SelectE/SelectE.tsx

```diff
@@ -0,0 +1,199 @@
+import { StyledInput } from './partials/StyledInput'
+import { StyledInputWrapper } from './partials/StyledInputWrapper'
+import { StyledLabel } from '../TextInputE/partials/StyledLabel'
+import { StyledOptions } from './partials/StyledOptions'
+import { StyledOptionsWrapper } from './partials/StyledOptionsWrapper'
+import { StyledOuterWrapper } from './partials/StyledOuterWrapper'
+import { StyledSelect } from './partials/StyledSelect'
+import { StyledTextInput } from '../TextInputE/partials/StyledTextInput'
+import { useFloatingElement } from '../../hooks'
+
+type SelectChildrenType =
+  | React.ReactElement<SelectEOptionProps>
+  | React.ReactElement<SelectEOptionProps>[]
+
+export interface SelectEProps extends ComponentWithClass {
+  /**
+   * Collection of options to display within the Select.
+   */
+  children: SelectChildrenType
+  /**
+   * Optional HTML `id` attribute to apply to the component.
+   */
+  id?: string
+  /**
+   * Toggles whether the component is disabled or not (preventing user interaction).
+   */
+  isDisabled?: boolean
+  /**
+   * Toggles whether the component is invalid or not.
+   */
+  isInvalid?: boolean
+  /**
+   * Text label to display within the component.
+   */
+  label: string
+  /**
+   * Optional handler invoked when the selected value changes.
+   */
+  onChange?: (value: string) => void
+  /**
+   * Optional HTML `value` attribute to apply to the component.
+   */
+  value?: string
+}
+
+function getChildBy(
+  children: SelectChildrenType,
+  propName: string,
+  value: string
+): React.ReactNode {
+  return React.Children.toArray(children).find((child: React.ReactNode) => {
+    return React.isValidElement(child) && child.props[propName] === value
+  })
+}
+
+function getSelectedValue(children: SelectChildrenType, value: string) {
+  const child = getChildBy(children, 'value', value)
+  return (child as React.ReactElement)?.props?.children
+}
+
+export const SelectE: React.FC<SelectEProps> = ({
+  children,
+  id = getId('select'),
+  isDisabled,
+  isInvalid,
+  label,
+  onChange,
+  value,
+  ...rest
+}) => {
+  const { targetElementRef, floatingElementRef, styles, attributes } =
+    useFloatingElement()
+  const [hasHover, setHasHover] = useState<boolean>(false)
+
+  return (
+    <Downshift
+      onChange={(selection) => {
+        if (onChange) {
+          const child = getChildBy(children, 'children', selection)
+          onChange((child as React.ReactElement)?.props?.value)
+        }
+      }}
+      initialSelectedItem={getSelectedValue(children, value)}
+      itemToString={(item) => item || ''}
+    >
+      {({
+        getInputProps,
+        getItemProps,
+        getMenuProps,
+        getRootProps,
+        getToggleButtonProps,
+        highlightedIndex,
+        inputValue,
+        isOpen,
+        openMenu,
+        toggleMenu,
+      }) => {
+        // @ts-ignore
+        const rootProps = getRootProps({}, { suppressRefError: true })
```

Comment:
The first argument to `getRootProps` is optional so can hopefully get rid of the `// @ts-ignore` by passing `undefined`:

```suggestion
        const rootProps = getRootProps(undefined, { suppressRefError: true })
```